rs/index/src/vector/fixed_file.rs

```diff
@@ -108,16 +108,26 @@ impl<T: ToBytes + Clone> FixedFileVectorStorage<T> {
         query: &[T],
         iterator: impl Iterator<Item = u64>,
         quantizer: &impl Quantizer<QuantizedT = T>,
-        invalidated_ids: &DashSet<u32>,
+        invalidated_ids: &RwLock<HashSet<u32>>,
         record_pages: bool,
     ) -> Result<IntermediateResult> {
         let mut result = vec![];
         let mut context = SearchContext::new(record_pages);
         let mut stats = SearchStats::new();
+        let invalidate_ids_clone = {
```

Comment:
It's better to make clear that this is an optional - I was confused by that for a bit
```suggestion
        let invalidated_ids_opt = {
```